GunShipAlpha/src/Enemy.java

```diff
@@ -1,41 +1,94 @@
 import java.awt.*;
+import java.awt.geom.Point2D;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
 
 class Enemy {
-    private int x, y;
-    private final int targetX, targetY;
+    private Point location;
+    private Point target;
     private int health = 2;
+    private Timer firingTimer;
+
     public Enemy(int x, int y, int targetX, int targetY) {
-        this.x = x;
-        this.y = y;
-        this.targetX = targetX;
-        this.targetY = targetY;
+        this.location = new Point(x, y);
+        this.target = new Point(targetX, targetY);
+        scheduleFiring();
     }
+    private void scheduleFiring() {
+        if (firingTimer != null) {
+            firingTimer.cancel();
+        }
+
+        firingTimer = new Timer();
+
+        TimerTask firingTask = new TimerTask() {
+            @Override
+            public void run() {
 
+                Projectile projectile = fireAtPlayer();
+                Game.gs.addProjectile(projectile);
+                scheduleFiring();
+            }
+        };
+        Random random = new Random();
+        long delay = 0 + random.nextInt(3000);
```

Comment:
0 + doesn't add any value (lol pun) here. it can just be 

long delay = random.nextInt(3000)

Reply:
fixed